lib/src/widgets/release_body.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:fwupd/fwupd.dart';
+import 'package:provider/provider.dart';
+import 'package:yaru_icons/yaru_icons.dart';
+import 'package:yaru_widgets/yaru_widgets.dart';
+
+import '../../device_model.dart';
+import '../../fwupd_x.dart';
+import 'confirmation_dialog.dart';
+import 'release_card.dart';
+
+class ReleaseBody extends StatelessWidget {
+  const ReleaseBody({
+    super.key,
+    required this.device,
+    required this.releases,
+    required this.onInstall,
+  });
+
+  final FwupdDevice device;
+  final List<FwupdRelease> releases;
+  final ValueChanged<FwupdRelease> onInstall;
+
+  @override
+  Widget build(BuildContext context) {
+    final model = context.watch<DeviceModel>();
+    final selected = model.selectedRelease;
+    final l10n = AppLocalizations.of(context);
+    final String action;
+    final String dialogText;
+
+    if (selected?.isDowngrade == true) {
+      action = l10n.downgrade;
+      dialogText = l10n.downgradeConfirm(
+        device.name,
+        device.version,
+        selected?.version,
+      );
+    } else if (selected?.isUpgrade == false) {
+      action = l10n.reinstall;
+      dialogText = l10n.reinstallConfirm(
+        device.name,
+        device.version,
+      );
+    } else {
+      action = l10n.upgrade;
+      dialogText = l10n.upgradeConfirm(
+        device.name,
+        device.version,
+        selected?.version,
+      );
+    }
+
+    return YaruPage(
+      children: [
+        Row(
+          children: [
+            IconButton(
+              onPressed: Navigator.of(context).pop,
+              icon: const Icon(YaruIcons.go_previous),
+            ),
+            Text(
+              '${device.name} ${device.version}',
+              style: Theme.of(context).textTheme.headlineSmall,
+            ),
+          ],
+        ),
+        Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          mainAxisSize: MainAxisSize.min,
+          children: releases
+              .map(
+                (release) => Flexible(
+                  child: ReleaseCard(
+                    release: release,
+                    selected: release == selected,
+                    onSelected: () =>
+                        context.read<DeviceModel>().selectedRelease = release,
+                  ),
+                ),
+              )
+              .toList(),
+        ),
+        ButtonBar(
```

Comment:
will the buttons get pushed out of the screen if there's a long list of releases?

Reply:
`YaruPage` wraps its content in a `SingleChildScrollView` so there won't be any overflow, but the buttons will end up on the very bottom - maybe it's better to fix the `ButtonBar` at the bottom of the screen and make only the list of releases scrollable?

Reply:
sounds good because release notes can be sometimes pretty long